### PR TITLE
Wsteama 124 add fonts to theming

### PR DESCRIPTION
**Pull request by jroebu14**
Supports WSTEAMA-124

**Overall change:**
Moves font faces and scripts from the legacy location into the new TypeScript ThemeProvider.

This is part of the theming work needed to support primitive components.

**Code changes:**

- moves scripts and font faces into `ThemeProvider` while also refactoring some var naming and updates all import paths
- updates Storybook's `.preview` so that fonts still load
- updates `ThemeProver` to enable services with variants to load unique themes. This is required for the different typography settings for Serbian Cyrillic and Latin

**Testing**
Requires regression testing for all services with variants and some without variants. Things to look out for are fonts not loading correctly or pages not rendering at all. All PR checks are passing with only slight changes to tests. ChromaticQA is flagging the normal level of diffs. If fonts were not loading then there might be more diffs.

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

File: src/app/Layouts/defaultPageWrapper.jsx [hunk 1]
```diff
@@ -30,8 +31,8 @@ const Content = styled.div`
 `;
 
 const PageWrapper = ({ children, pageData, status }) => {
-  const { fonts: fontFunctions, service } = useContext(ServiceContext);
-  const fonts = fontFunctions.map(getFonts => getFonts());
+  const { service } = useContext(ServiceContext);
+  const { variant } = useContext(RequestContext);
```
Comment (amoore108) [inline src/app/Layouts/defaultPageWrapper.jsx]: I think `variant` is available through `ServiceContext` as well. Not sure why its exposed through both contexts though 🤔 

EDIT: Actually `RequestContext` can provide `service` too. Out of scope for this I guess, but we should probably consolidate to one or the other.

Comment (jroebu14) [reply src/app/Layouts/defaultPageWrapper.jsx]: Ah I thought it was but when developing I'm sure it was `undefined` when pulled from ServiceContext. I've tested just now and it does work with ServiceContext so I'll use that. Not sure what happened. Thanks!

**Review by amoore108 — APPROVED** [review]
Looks great 🙌🏼

File: src/app/components/ThemeProvider/README.md [hunk 2]
```diff
@@ -23,3 +23,24 @@ Add `ThemeProvider` to the top level of your app and access the theme with props
...
 ```
+
+## Typography scripts
+
+These are `font-sizes` and `line-heights` for a specific service's GEL size (e.g. `Long Primer`).
+
+Simorgh supports typography for a number of other scripts. Below are all supported scripts and the services that use them.
+
+| Group Name          | Services                                                                                  | Rationale                                                                                                   |
+| ------------------- | ----------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------- | ------------------------------------------------- | -------------------------------------------------------------------------- |
```
Comment (DarioR01) [inline src/app/components/ThemeProvider/README.md]: Very Minor adjustment needed here, When I preview the file it looks like this:
Screenshot 2022-08-23 at 15 49 38

Comment (DarioR01) [reply src/app/components/ThemeProvider/README.md]:
```suggestion
| ------------------- | ---------------------------- |------------------------------------- |
```

**Review by DarioR01 — APPROVED** [review]

File: src/app/components/ThemeProvider/README.md (hunk 2 above)
Comment (jroebu14) [reply src/app/components/ThemeProvider/README.md]: Thanks, @DarioR01. Sorted!

**Review by ryanmccombe — DISMISSED** [review]

**Review by RetiredBlob — APPROVED** [review]

File: src/app/Layouts/defaultPageWrapper.jsx (hunk 1 above)
Comment (ExcitedBlob) [reply src/app/Layouts/defaultPageWrapper.jsx]: Seen some scenarios where `variant` is undefined or different depending on which context you get it from

This only happens on tests or storybook though where people are importing `<ServiceContextProvider>` and `<RequestContextProvider>` and then passing their own arguments to them - eg, they provide a `variant` to one component, but not the other

**Pull request merged**